plugins/reports/counter/classes/CounterReport.inc.php

```diff
@@ -0,0 +1,301 @@
+ *
+ * @brief A COUNTER report, base class
+ */
+require_once(dirname(dirname(__FILE__)).'/classes/COUNTER/COUNTER.php');
+
+define('COUNTER_EXCEPTION_WARNING', 0);
+define('COUNTER_EXCEPTION_ERROR', 1);
+define('COUNTER_EXCEPTION_PARTIAL_DATA', 4);
+define('COUNTER_EXCEPTION_NO_DATA', 8);
+define('COUNTER_EXCEPTION_BAD_COLUMNS', 16);
+define('COUNTER_EXCEPTION_BAD_FILTERS', 32);
+define('COUNTER_EXCEPTION_BAD_ORDERBY', 64);
+define('COUNTER_EXCEPTION_BAD_RANGE', 128);
+define('COUNTER_EXCEPTION_INTERNAL', 256);
+
+define('COUNTER_CLASS_PREFIX', 'CounterReport');
+
+// COUNTER as of yet is not internationalized and requires English constants
+define('COUNTER_LITERAL_ARTICLE', 'Article');
+define('COUNTER_LITERAL_JOURNAL', 'Journal');
+define('COUNTER_LITERAL_PROPRIETARY', 'Proprietary');
+
+class CounterReport {
+
+	var $parentPluginCategory;
+	var $parentPluginName;
+	
+	var $_release;
+	
+	var $_errors;
+	
+	/**
+	 * Constructor
+	 * @param string $parentPluginCategory
+	 * @param string $parentPluginName
+	 */
+	function CounterReport($parentPluginCategory, $parentPluginName) {
+		$this->parentPluginCategory = $parentPluginCategory;
+		$this->parentPluginName = $parentPluginName;
+		$this->_release = $this->getParentPlugin()->getCurrentRelease();
+	}
+	
+
+	/**
+	 * Get the parent plugin
+	 * @return object
+	 */
+	function &getParentPlugin() {
+		$plugin =& PluginRegistry::getPlugin($this->parentPluginCategory, $this->parentPluginName);
+		return $plugin;
+	}
+
+	/**
+	 * Get the COUNTER Release
+	 * @return $string
+	 */
+	function getRelease() {
+		return $this->_release;
+	}
+
+	/**
+	 * Get the report code
+	 * @return $string
+	 */
+	function getCode() {
+		return substr(get_class($this), strlen(COUNTER_CLASS_PREFIX));
+	}
+
+	/**
+	 * Get the COUNTER metric type for an Statistics file type
+	 * @param $filetype string
+	 * @return string
+	 */
+	function getKeyForFiletype($filetype) {
+		switch ($filetype) {
+			case STATISTICS_FILE_TYPE_HTML:
+				$metricTypeKey = 'ft_html';
+				break;
+			case STATISTICS_FILE_TYPE_PDF:
+				$metricTypeKey = 'ft_pdf';
+				break;
+			case STATISTICS_FILE_TYPE_OTHER:
+			default:
+				$metricTypeKey = 'other';
+		}
+		return $metricTypeKey;
+	}
+
+	/**
+	 * Abstract method must be implemented in the child class
+	 * Get the report title
+	 * @return $string
+	 */
+	function getTitle() {
+		assert(false);
+	}
+
+	/*
+	 * Convert an OJS metrics request to COUNTER ReportItems
+	 * Abstract method must be implemented by subclass
+	 * @param $columns string|array column (aggregation level) selection
+	 * @param $filters array report-level filter selection
+	 * @param $orderBy array order criteria
+	 * @param $range null|DBResultRange paging specification
+	 * @see ReportPlugin::getMetrics for more details on parameters
+	 * @return array COUNTER\ReportItem array
+	 */
+	function getReportItems($columns = array(), $filters = array(), $orderBy = array(), $range = null) {
+		assert(false);
```

Comment:
since you're already using some PHP 5 stuff, you can declare this class and method abstracts... But not sure about using PHP 5 things like this in ojs 2.x @asmecher are we doing this now?